generalchannel/__init__.py

```diff
@@ -5,5 +5,6 @@
 )
 
 
-def setup(bot):
-    bot.add_cog(GeneralChannel(bot))
+async def setup(bot):
+    cog = GeneralChannel(bot)
+    await bot.add_cog(cog)
```

Comment:
Why adding an variable here?
```suggestion
async def setup(bot):
    await bot.add_cog(GeneralChannel(bot))
```